vnext/Microsoft.ReactNative/ReactNativeWin32App.cpp

```diff
@@ -0,0 +1,69 @@
+#include "pch.h"
+#include "ReactNativeWin32App.h"
+#include "ReactNativeWin32App.g.cpp"
+#include "winrt/Microsoft.UI.Composition.h"
+#include "winrt/Microsoft.UI.Content.h"
+#include "winrt/Microsoft.UI.Interop.h"
+#include "winrt/Microsoft.UI.Windowing.h"
+
+namespace winrt::Microsoft::ReactNative::implementation {
+ReactNativeWin32App::ReactNativeWin32App() {}
+
+ReactNativeWin32App::~ReactNativeWin32App() {
+  m_desktopChildSiteBridge.Close();
+  m_desktopChildSiteBridge = nullptr;
+
+  // Destroy all Composition objects
+  m_compositor.Close();
+  m_compositor = nullptr;
+}
+
+void ReactNativeWin32App::AppWindow(winrt::Microsoft::UI::Windowing::AppWindow const &appWindow) {
+  m_appWindow = appWindow;
+}
+
+winrt::Microsoft::UI::Windowing::AppWindow ReactNativeWin32App::AppWindow() {
+  return m_appWindow;
+}
+
+void ReactNativeWin32App::Compositor(winrt::Microsoft::UI::Composition::Compositor const &compositor) {
+  if (compositor == nullptr) {
+    m_compositor = winrt::Microsoft::UI::Composition::Compositor();
```

Comment:
I don't think this logic should be here. If they pass in nullptr, set it to nullptr. Let build set it to something new if necessary.